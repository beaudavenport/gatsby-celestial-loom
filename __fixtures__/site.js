export default {
  site: {
    siteMetadata: {
      title: 'The Celestial Loom',
    },
  },
};
