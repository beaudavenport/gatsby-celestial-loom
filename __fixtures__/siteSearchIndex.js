export default {
  siteSearchIndex: {
    index: null,
  },
};
