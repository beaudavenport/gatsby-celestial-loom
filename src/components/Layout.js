import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { StaticQuery, graphql, Link } from 'gatsby';
import {
  NavigationDrawer, FontIcon, Media, CardText, ListItem, Drawer,
} from 'react-md';
import HeaderSearch from './HeaderSearch';
import Nikki from '../images/nikki.jpg';

import './layout.scss';

const navItems = [
  { to: '/posts', children: (<p style={{ color: 'white', fontSize: '14px' }}>Posts</p>) },
  { to: '/events', children: (<p style={{ color: 'white', fontSize: '14px' }}>Events</p>) },
  { to: '/services', children: (<p style={{ color: 'white', fontSize: '14px' }}>Services</p>) },
  { to: '/toolbox', children: (<p style={{ color: 'white', fontSize: '14px' }}>Astro Toolbox</p>) },
];

const DrawerHeader = () => (
  <div style={{
    width: '80%', margin: '0 auto',
  }}
  >
    <div style={{ width: '60%', margin: '0 auto', padding: '20px' }}>
      <Media aspectRatio="1-1" style={{ borderRadius: '50%', border: '2px solid #ff8013' }}>
        <img src={Nikki} alt="at da club" />
      </Media>
    </div>
    <CardText>
      <p style={{ fontSize: 'smaller', color: '#ff8013', textAlign: 'center' }}>
          Nikki Davenport, astrological consultant, has been a professional
          astrologer for over 35 years.
      </p>
    </CardText>
  </div>
);

const DrawerTitle = () => (
  <div style={{ paddingTop: '10px', paddingLeft: '10px' }}>
    <h1 className="md-headline" style={{ fontFamily: 'Berkshire Swash', fontStyle: 'italic', color: 'white' }}>The Celestial Loom</h1>
  </div>
);

const Layout = ({ children }) => {
  const { mobile, tablet } = Drawer.getCurrentMedia(Drawer.defaultProps);
  let defaultMedia = 'desktop';
  if (mobile) {
    defaultMedia = 'mobile';
  } else if (tablet) {
    defaultMedia = 'tablet';
  }
  return (
    <StaticQuery
      query={graphql`
          query SiteTitleQuery {
            site {
              siteMetadata {
                title
              }
            }
          }
        `}
      render={data => (
        <NavigationDrawer
          desktopDrawerType={NavigationDrawer.DrawerTypes.FULL_HEIGHT}
          tabletDrawerType={NavigationDrawer.DrawerTypes.TEMPORARY}
          mobileDrawerType={NavigationDrawer.DrawerTypes.TEMPORARY}
          position="left"
          drawerStyle={{ backgroundColor: 'rgb(47, 47, 47)' }}
          defaultVisible={false}
          temporaryIcon={(<FontIcon iconClassName="material-icons">menu</FontIcon>)}
          navStyle={{ height: '30%', backgroundColor: 'rgb(47, 47, 47)' }}
          drawerChildren={<DrawerHeader />}
          toolbarTitle={defaultMedia === 'desktop' ? null : 'The Celestial Loom'}
          toolbarActions={<HeaderSearch />}
          drawerHeader={defaultMedia !== 'desktop' ? null : (<DrawerTitle />)}
          toolbarTitleStyle={{ fontFamily: 'Berkshire Swash', fontStyle: 'italic' }}
          drawerTitleStyle={{ fontFamily: 'Berkshire Swash', fontStyle: 'italic' }}
          toolbarSingleColor
          toolbarStyle={{ backgroundColor: 'rgb(47, 47, 47)' }}
          defaultMedia={defaultMedia}
          navItems={navItems.map(({ divider, subheader, ...route }) => {
            if (divider || subheader) {
              return { divider, subheader, ...route };
            }

            return (
              <ListItem
                tileStyle={{ color: 'white', textAlign: 'left' }}
                primaryText={route.children}
                style={{ padding: '0px 10px', color: 'white' }}
                component={Link}
                to={route.to}
              />
            );
          })}
        >
          <Helmet
            title={data.site.siteMetadata.title}
            meta={[
              { name: 'description', content: 'Sample' },
              { name: 'keywords', content: 'sample, something' },
            ]}
          >
            <html lang="en" />
            <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.2/jquery.min.js" />
            <script src="https://cdn.snipcart.com/scripts/2.0/snipcart.js" data-api-key="Y2NiNmUxNzEtMGNlZS00NTlkLTg1NDEtZjJjMGRjMWNjZThjNjM2ODU2Njg0NDE5MzU2MDg3" id="snipcart" />
            <link href="https://cdn.snipcart.com/themes/2.0/base/snipcart.min.css" rel="stylesheet" type="text/css" />
          </Helmet>
          {children}
          <div>
            <hr />
            <h6 style={{ textAlign: 'center' }}>Copyright 2019, The Celestial Loom. Cover photography by Photo by Anastasia Dulgier on Unsplash. Built with Gatsby and Netflify CMS.</h6>
          </div>
        </NavigationDrawer>
      )}
    />
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
