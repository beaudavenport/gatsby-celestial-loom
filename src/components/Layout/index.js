import '../main.scss';
import { Grid, GridCell } from "@react-md/utils"
import PropTypes from 'prop-types';
import React from 'react';
import { HomeSVGIcon, InfoSVGIcon, CreateSVGIcon, EventSVGIcon, ShoppingCartSVGIcon, BuildSVGIcon } from "@react-md/material-icons";
import {
  DEFAULT_PHONE_LAYOUT,
  Layout,
  useLayoutNavigation,
  Configuration
} from "@react-md/layout";
import { Link } from 'gatsby';

import Footer from '../Footer';
import HeaderCart from '../HeaderCart';
import HeaderSearch from '../HeaderSearch';
import SEOAndScripts from '../SEOAndScripts';

const navItems = {
  "/": { itemId: "/", children: "Home", to: '/', parentId: null, leftAddon: <HomeSVGIcon fill="white" stroke="white" /> },
  "/info": { itemId: "/getting-started", children: "Getting Started", to: '/getting-started', parentId: null, leftAddon: <InfoSVGIcon fill="white" stroke="white" /> },
  "/posts": { itemId: "/posts", children: "Posts", to: '/posts', parentId: null, leftAddon: <CreateSVGIcon fill="white" stroke="white" /> },
  "/events": { itemId: "/events", children: "Events", to: '/events', parentId: null, leftAddon: <EventSVGIcon fill="white" stroke="white" /> },
  "/services": { itemId: "/services", children: "Services", to: '/services', parentId: null, leftAddon: <ShoppingCartSVGIcon fill="white" stroke="white" /> },
  "/toolbox": { itemId: "/toolbox", children: "Astro Toolbox", to: '/toolbox', parentId: null, leftAddon: <BuildSVGIcon fill="white" stroke="white" /> }
}

const SiteLayout = ({
  jumbotron,
  children,
  sidebarChildren,
  title,
  seoTitle,
  seoDescription,
  seoImage,
  seoPathname,
  isArticle,
}) => {

  return (
    <Configuration>
      <Layout
        id="configurable-layout"
        title={title}
        navHeaderTitle="The Celestial Loom"
        phoneLayout={DEFAULT_PHONE_LAYOUT}
        tabletLayout={"temporary"}
        landscapeTabletLayout={"temporary"}
        desktopLayout={"temporary"}
        treeProps={{
          ...useLayoutNavigation(navItems, window.location.pathname, Link),
          style: { backgroundColor: "#2f2f2f", color: "white"}
        }}
        appBarProps={{
          children: 
            <div className="header-icons-container">
            <HeaderCart />
            <HeaderSearch />
          </div>
        }}
      >
        <SEOAndScripts
          title={seoTitle}
          description={seoDescription || ''}
          image={seoImage}
          pathname={seoPathname}
          article={isArticle}
        />
        {jumbotron}
        <Grid style={{ maxWidth: 1100, margin: "0 auto" }}>
          <GridCell className="main-contents" colSpan={8}>
            {children}
          </GridCell>
          <GridCell colSpan={4} className="sidebar-contents">
            {sidebarChildren}
          </GridCell>
        </Grid>
        <Footer />
      </Layout>
    </Configuration>
  )
};

SiteLayout.propTypes = {
  jumbotron: PropTypes.node,
  children: PropTypes.node.isRequired,
  sidebarChildren: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  seoTitle: PropTypes.string,
  seoDescription: PropTypes.string,
  seoImage: PropTypes.string,
  seoPathname: PropTypes.string,
  isArticle: PropTypes.bool,
};

SiteLayout.defaultProps = {
  jumbotron: null,
  seoTitle: null,
  seoDescription: null,
  seoImage: null,
  seoPathname: null,
  isArticle: false,
};

export default SiteLayout;
