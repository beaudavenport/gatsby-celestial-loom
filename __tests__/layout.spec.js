import { StaticQuery } from 'gatsby';
import React from 'react';

import renderer from 'react-test-renderer';

import Layout from '../src/components/Layout';
import site from '../__fixtures__/site';
import siteSearchIndex from '../__fixtures__/siteSearchIndex';

beforeEach(() => {
  global.window.matchMedia = jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
  }));

  StaticQuery.mockImplementation(({ render }) => render({
    ...site,
    ...siteSearchIndex,
  }));
});

describe('Layout', () => {
  it('should render correctly with no children', () => {
    const tree = renderer
      .create(<Layout />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
