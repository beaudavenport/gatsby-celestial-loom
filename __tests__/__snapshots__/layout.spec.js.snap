// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly with no children 1`] = `
<div>
  <header
    className="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed"
    style={
      Object {
        "backgroundColor": "rgb(47, 47, 47)",
      }
    }
  >
    <button
      className="md-btn md-btn--icon md-pointer--hover md-inline-block md-btn--toolbar md-toolbar--action-left"
      disabled={false}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-ink-container"
      />
      <i
        className="md-icon material-icons md-text--inherit"
      >
        menu
      </i>
    </button>
    <div
      className="md-cell--right md-toolbar--action-right"
    >
      <div
        className="md-btn--toolbar header-icons-container"
      >
        <div
          className="header-search-container"
        >
          <button
            className="md-btn md-pointer--hover md-text md-inline-block"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
            type="button"
          >
            <div
              className="md-ink-container"
            />
            <div
              className="md-badge-container md-inline-block"
            >
              <i
                className="md-icon material-icons"
                style={
                  Object {
                    "fontSize": "2rem",
                  }
                }
              >
                shopping_cart
              </i>
            </div>
          </button>
        </div>
        <div
          className="header-search-container"
        >
          <button
            className="md-btn md-pointer--hover md-text md-inline-block"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
            type="button"
          >
            <div
              className="md-ink-container"
            />
            <i
              className="md-icon material-icons"
              style={
                Object {
                  "fontSize": "2rem",
                }
              }
            >
              search
            </i>
          </button>
        </div>
      </div>
    </div>
  </header>
  <main
    className="md-navigation-drawer-content md-navigation-drawer-content--inactive md-transition--acceleration md-toolbar-relative"
    id="main-content"
    tabIndex={-1}
  >
    <div
      className="md-grid md-grid--no-spacing"
      style={
        Object {
          "maxWidth": 1100,
        }
      }
    >
      <div
        className="md-cell md-cell--8 main-contents"
      />
      <div
        className="md-cell md-cell--4 md-cell--8-tablet sidebar-contents"
      />
    </div>
    <div
      className="md-grid footer--grid"
    >
      <div
        className="md-cell md-cell--6 md-cell--8-tablet"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <div
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <img
              alt="Celestial Loom Logo"
              src="File contents (usually images)"
              style={
                Object {
                  "width": 50,
                }
              }
            />
          </div>
          <h3
            className="footer--title"
          >
            The Celestial Loom
          </h3>
        </div>
        <div
          style={
            Object {
              "padding": 10,
            }
          }
        >
          <p
            className="footer--text"
          >
            Â©
             Copyright 2019, Nikki Davenport.
          </p>
          <p
            className="footer--text"
          >
            Cover photography by Photo by Anastasia Dulgier on Unsplash.
          </p>
          <p
            className="footer--text"
          >
            Built with Gatsby and Netflify CMS.
          </p>
        </div>
      </div>
      <div
        className="md-cell md-cell--6 md-cell--8-tablet"
      >
        <div
          className="md-grid"
        >
          <div
            className="md-cell md-cell--6 md-cell--4-tablet"
          >
            <p
              className="subheader"
            >
              Links
            </p>
            <ul
              className="footer--list"
            >
              <li
                className="footer--line-item"
              >
                <a
                  href="/home/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Home
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/about-nikki/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  About Nikki
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/getting-started/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Getting Started
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/posts/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Posts
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/events/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Events
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/posts/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Services
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/posts/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Astro Toolbox
                </a>
              </li>
            </ul>
          </div>
          <div
            className="md-cell md-cell--6 md-cell--4-tablet"
          >
            <p
              className="subheader"
            >
              Contact
            </p>
            <ul
              className="footer--list"
            >
              <li>
                <div
                  className="footer--link"
                >
                  <i
                    className="md-icon material-icons"
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    email
                  </i>
                  <a
                    className="footer--link__link"
                    href="mailto: nikiastro@att.net"
                  >
                    Email Nikki
                  </a>
                </div>
              </li>
              <li>
                <div
                  className="footer--link"
                >
                  <svg
                    fill="white"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm3 8h-1.35c-.538 0-.65.221-.65.778v1.222h2l-.209 2h-1.791v7h-3v-7h-2v-2h2v-2.308c0-1.769.931-2.692 3.029-2.692h1.971v3z"
                    />
                  </svg>
                  <a
                    className="footer--link__link"
                    href="https://www.facebook.com/celestialloom/"
                  >
                    Facebook
                  </a>
                </div>
              </li>
              <li>
                <div
                  className="footer--link"
                >
                  <svg
                    fill="white"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm6.066 9.645c.183 4.04-2.83 8.544-8.164 8.544-1.622 0-3.131-.476-4.402-1.291 1.524.18 3.045-.244 4.252-1.189-1.256-.023-2.317-.854-2.684-1.995.451.086.895.061 1.298-.049-1.381-.278-2.335-1.522-2.304-2.853.388.215.83.344 1.301.359-1.279-.855-1.641-2.544-.889-3.835 1.416 1.738 3.533 2.881 5.92 3.001-.419-1.796.944-3.527 2.799-3.527.825 0 1.572.349 2.096.907.654-.128 1.27-.368 1.824-.697-.215.671-.67 1.233-1.263 1.589.581-.07 1.135-.224 1.649-.453-.384.578-.87 1.084-1.433 1.489z"
                    />
                  </svg>
                  <a
                    className="footer--link__link"
                    href="https://twitter.com/nikiastro"
                  >
                    Twitter
                  </a>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
